packages/next/server/view-render.tsx

```diff
@@ -98,57 +98,55 @@ function preloadDataFetchingRecord(
   return record
 }
 
-function createFlightHook() {
-  return (
-    writable: WritableStream<Uint8Array>,
-    id: string,
-    req: ReadableStream<Uint8Array>,
-    bootstrap: boolean
-  ) => {
-    let entry = rscCache.get(id)
-    if (!entry) {
-      const [renderStream, forwardStream] = readableStreamTee(req)
-      entry = createFromReadableStream(renderStream)
-      rscCache.set(id, entry)
-
-      let bootstrapped = false
-      const forwardReader = forwardStream.getReader()
-      const writer = writable.getWriter()
-      function process() {
-        forwardReader.read().then(({ done, value }) => {
-          if (bootstrap && !bootstrapped) {
-            bootstrapped = true
-            writer.write(
-              encodeText(
-                `<script>(self.__next_s=self.__next_s||[]).push(${JSON.stringify(
-                  [0, id]
-                )})</script>`
-              )
+function useFlightResponse(
+  writable: WritableStream<Uint8Array>,
+  id: string,
+  req: ReadableStream<Uint8Array>
+) {
+  let entry = rscCache.get(id)
+  if (!entry) {
+    const [renderStream, forwardStream] = readableStreamTee(req)
+    entry = createFromReadableStream(renderStream)
+    rscCache.set(id, entry)
+
+    let bootstrapped = false
+    const forwardReader = forwardStream.getReader()
+    const writer = writable.getWriter()
+    function process() {
+      forwardReader.read().then(({ done, value }) => {
+        if (!bootstrapped) {
+          bootstrapped = true
+          writer.write(
+            encodeText(
+              `<script>(self.__next_s=self.__next_s||[]).push(${JSON.stringify([
```

Comment:
shall we update JSON.stringify with escapeJSONForFlightScript as well?

Reply:
Yeah probably in another PR. I only applied #36989 here.

Reply:
Actually this branch is out-of-date and checks will be started again anyways. I can push the change here.